packages/trpc/server/routers/viewer/organizations/create.handler.ts

```diff
@@ -38,8 +47,279 @@ const getIPAddress = async (url: string): Promise<string> => {
+  isOrganizationConfigured: boolean;
+  isOrganizationAdminReviewed: boolean;
+  autoAcceptEmail: string;
+  seats: number | null;
+  pricePerSeat: number | null;
+  isPlatform: boolean;
+};
+
+const persistOrganization = async (orgData: OrgData, orgOwnerEmail: string) => {
+  const orgOwner = await prisma.user.findUnique({
+    where: {
+      email: orgOwnerEmail,
+    },
+  });
+
+  if (!orgOwner) {
+    // do stuff for not org owner
+    const data = await OrganizationRepository.createWithNonExistentOwner({
+      orgData,
+      owner: {
+        email: orgOwnerEmail,
+      },
+    });
+
+    orgOwner = data.orgOwner;
+
+    const { organization, ownerProfile } = data;
+
+    const translation = await getTranslation(input.language ?? "en", "common");
+
+    await sendEmailVerification({
+      email: orgOwnerEmail,
+      language: ctx.user.locale,
+      username: ownerProfile.username || "",
+    });
+
+    await sendEmailAndEnrichProfile(false);
+  } else {
+    // do stuff for org owner
+    // If we are making the loggedIn user the owner of the organization and he is already a part of an organization, we don't allow it because multi-org is not supported yet
+    const isLoggedInUserOrgOwner = orgOwner.id === loggedInUser.id;
+    if (ctx.user.profile.organizationId && isLoggedInUserOrgOwner) {
+      throw new TRPCError({ code: "FORBIDDEN", message: "User is part of an organization already" });
+    }
+
+    const nonOrgUsernameForOwner = orgOwner.username || "";
+    const { organization, ownerProfile } = await OrganizationRepository.createWithExistingUserAsOwner({
+      orgData,
+      owner: {
+        id: orgOwner.id,
+        email: orgOwnerEmail,
+        nonOrgUsername: nonOrgUsernameForOwner,
+      },
+    });
+
+    // this will be standalone line
+    if (!organization.id) throw Error("User not created");
+
+    // this repeats for both if else cases
+    await sendEmailAndEnrichProfile(true);
+  }
+};
+
+const checkOrgOwnerCreationRequirements = async () => {
+  // check for admin
+  const isAdmin = checkUserIsAdminOrThrow();
+
+  // check for published team
+  checkOrgPublishedTeams();
+
+  // check for url is taken or not
+  await isOrgSlugTaken();
+
+  return isAdmin;
+};
+
+const createOrgUser = async () => {
+  const loggedInUser = await checkLoginStatus();
+
+  const isAdmin = await checkOrgOwnerCreationRequirements();
+
+  // this is also included for createPlatformOrg fn
+  const availability = getAvailabilityFromSchedule(DEFAULT_SCHEDULE);
+
+  // this is also included for createPlatformOrg fn
+  await configureOrg();
+
+  // this is also included for createPlatformOrg fn
+  await persistOrganization();
+};
+
+const createPlatformUser = async () => {
+  const loggedInUser = await checkLoginStatus();
+
+  // this is also included for createPlatformOrg fn
+  const availability = getAvailabilityFromSchedule(DEFAULT_SCHEDULE);
+
+  // this is also included for createPlatformOrg fn
+  await configureOrg();
```

Comment:
why might not need everything happening in configureOrg to happen for platform, like the creation of the domain for example